src/Platform/Microsoft.Testing.Platform/OutputDevice/Terminal/NonAnsiTerminal.cs

```diff
@@ -116,20 +63,32 @@ public void HideCursor()
         // nop
     }
 
+    // TODO: Refactor NonAnsiTerminal and AnsiTerminal such that we don't need StartUpdate/StopUpdate.
+    // It's much better if we use lock C# keyword instead of manually calling Monitor.Enter/Exit
+    // Using lock also ensures we don't accidentally have `await`s in between that could cause Exit to be on a different thread.
     public void StartUpdate()
     {
         if (_isBatching)
         {
             throw new InvalidOperationException(PlatformResources.ConsoleIsAlreadyInBatchingMode);
         }
 
-        _stringBuilder.Clear();
+        bool lockTaken = false;
+        // SystemConsole.ConsoleOut is set only once in static ctor.
+        // So we are sure we will be doing Monitor.Exit on the same instance.
```

Comment:
I think I need to doc this more, and mention that it's to avoid interleaving with user's calls to `Console.WriteLine`, which will internally lock on Console.Out.

Reply:
I'll merge this as-is for now to unblock main CI that is currently blocked so we unblock DotNet sdk

Reply:
I already extended the comment so this comment wasn't valid anymore actually :)